src/AbstractCliApplication.php

```diff
@@ -51,6 +53,9 @@ public function __construct(Input\Cli $input = null, Registry $config = null, Cl
 		// @codeCoverageIgnoreEnd
 
 		$this->output = ($output instanceof CliOutput) ? $output : new Cli\Output\Stdout;
+		$this->getContainer()->set('Joomla\\Application\\Cli\\CliOutput', $this->output);
+		$this->getContainer()->alias('CliOutput', 'Joomla\\Application\\Cli\\CliOutput');
+		$this->getContainer()->alias('output', 'Joomla\\Application\\Cli\\CliOutput');
```

Comment:
What's the benefit to making every class var an object in the DIC?  This also feels like a major step backward.  It almost encourages us to stop using DI and just use our container as a service locator for everything.